keps/sig-api-machinery/4006-transition-spdy-to-websockets/README.md

```diff
@@ -750,10 +935,12 @@ Any change of default behavior may be surprising to users or break existing
 automations, so be extremely careful here.
 -->
 
-Enabling the feature gate on the API Server will allow the streaming mechanism
+For each of the two streaming subprotocols: `RemoteCommand` and `PortForward`,
+enabling the feature gate on the API Server will allow the streaming mechanism
 to be WebSockets instead of SPDY for communication between `kubectl` and the API
-Server. The `kubectl` client must also have the KUBECTL_REMOTE_COMMAND_WEBSOCKETS
-environment variable set to **ON**, so it will request the newer WebSockets streaming
+Server. Additionally, the `kubectl` client must also have either the
+KUBECTL_REMOTE_COMMAND_WEBSOCKETS or KUBECTL_PORT_FORWARD_WEBSOCKETS environment
+variables set to **ON**, so this client will request the newer WebSockets streaming
```

Comment:
I have added a separate **NOTE** to this section to clarify the separate maturity levels of these two sets of feature gates as of v1.30. And as @jpbetz suggested, I have also updated the `Graduation Criteria` section to also make it clear which features are graduating at which version. Also, the `Implementation History` section describes different maturity levels. I think this it is clearer now. Please let me know what you think.

With the separate set of feature gates, each of these subprotocols can be enabled or disabled separately. It would be possible (but not likely) that `PortForward` could be enabled while `RemoteCommand` is disabled.

In previous releases we discussed the possibility of separate KEP's. But each time it was determined it would be better to keep these two development tracks together in the same KEP. It should also be noted, that if we progress as planned (`PortForward` to Beta in v1.31, while `RemoteCommand` stays Beta in v1.31), there will be no difference in the maturity levels in this next v1.31 release.

Reply:
The additions are great, I'd still try to change the wording in:

```
 Additionally, the `kubectl` client must also have either the
KUBECTL_REMOTE_COMMAND_WEBSOCKETS or KUBECTL_PORT_FORWARD_WEBSOCKETS environment
variables set to **ON**, so this client will request the newer WebSockets streaming
```
to something more explicit, iow:
```
Additionally, the `kubectl` client must also have the KUBECTL_REMOTE_COMMAND_WEBSOCKETS 
environment variable set to ON for exec, cp, and attach commands.  
KUBECTL_PORT_FORWARD_WEBSOCKETS environment
variables set to **ON** for port-forward command.
```

Reply:
Similarly in the kube-apiserver section:
Instead of :
```
For each of the two streaming subprotocols: `RemoteCommand` and `PortForward`,
enabling the feature gate on the API Server will allow the streaming mechanism
to be WebSockets instead of SPDY for communication between `kubectl` and the API
```
write
```
For each of the two streaming subprotocols: `RemoteCommand` (such as /exec and /attach APIs) 
and `PortForward` (for /portforward), enabling the respective feature gate on the API Server will 
allow the streaming mechanism to be WebSockets instead of SPDY for communication between 
`kubectl` and the API
```

Reply:
The wording has been updated. Please let me know what you think.